### DISCUSSION TITLE
Removing Italian Stop Words

### DISCUSSION THREAD

**Discussion by AlessandroZanotta98** [MaartenGr/BERTopic#1860] [Q&A] [answered]
Hi everyone! 
I am doing topic modelling on italian reviews.
 I would like to know if removing stop words ahead of the creation of embeddings with SentenceTransformer (the model that I am using is"nickprock/sentence-bert-base-italian-uncased") changes in an important way the quality of the final embeddings.

Thank you in advance and have a nice day

**Comment by MaartenGr** [comment, marked as answer]
Generally, you would want to prevent removing stopwords from your raw documents. The only preprocessing they need is removing strange artifacts like HTML code if you for example scraped the data. So information not related to the content of the document. Thus, you generally do not want to remove stopwords **before** clustering your documents.

However, removing stopwords **after** clustering your documents for the topic representations might be helpful. You can find more about removing stop words directly [here](https://maartengr.github.io/BERTopic/getting_started/vectorizers/vectorizers.html) or you can use [KeyBERTInspired](https://maartengr.github.io/BERTopic/getting_started/representation/representation.html#keybertinspired) or [MMR](https://maartengr.github.io/BERTopic/getting_started/representation/representation.html#maximalmarginalrelevance) which do this indirectly. 

Also, following the [best practices guide](https://maartengr.github.io/BERTopic/getting_started/best_practices/best_practices.html) often helps get a sense of these things.

**Reply by AlessandroZanotta98** [reply]
Thank you for the answer Maarten! Congratulations again for the beautiful algorithm.
